github/data_source_github_repository_branches_test.go

```diff
@@ -54,4 +54,83 @@ func TestAccGithubRepositoryBranchesDataSource(t *testing.T) {
 			testCase(t, organization)
 		})
 	})
+
+	t.Run("manages branches of a new repository with filtering", func(t *testing.T) {
+		repoName := fmt.Sprintf("tf-acc-test-branches-%s", acctest.RandString(5))
+		config := fmt.Sprintf(`
+			resource "github_repository" "test" {
+				name      = "%s"
+				auto_init = true
+			}
+
+			resource "github_branch" "test" {
+				repository = github_repository.test.id
+				branch     = "test"
+			}
+
+			resource "github_branch_protection_v3" "test" {
+		  	repository = github_repository.test.name
+		  	branch     = "test"
+		  	depends_on = [github_branch.test]
+		  }
+		 `, repoName)
+
+		config2 := config + `
+			data "github_repository_branches" "test" {
+				repository = github_repository.test.name
+			}
+
+			data "github_repository_branches" "protected" {
+				repository              = github_repository.test.name
+				only_protected_branches = true
+			}
+
+			data "github_repository_branches" "non_protected" {
+				repository                  = github_repository.test.name
+				only_non_protected_branches = true
+			}
+		`
+
+		const resourceName = "data.github_repository_branches.test"
+		const protectedResourceName = "data.github_repository_branches.protected"
+		const nonProtectedResourceName = "data.github_repository_branches.non_protected"
+		check := resource.ComposeTestCheckFunc(
+			resource.TestCheckResourceAttr(resourceName, "branches.#", "2"),
+			resource.TestCheckResourceAttr(protectedResourceName, "branches.#", "1"),
+			resource.TestCheckResourceAttr(protectedResourceName, "branches.0.name", "test"),
+			resource.TestCheckResourceAttr(protectedResourceName, "branches.0.protected", "true"),
+			resource.TestCheckResourceAttr(nonProtectedResourceName, "branches.#", "1"),
+			resource.TestCheckResourceAttr(nonProtectedResourceName, "branches.0.name", "main"),
+			resource.TestCheckResourceAttr(nonProtectedResourceName, "branches.0.protected", "false"),
+		)
+
+		testCase := func(t *testing.T, mode string) {
+			resource.Test(t, resource.TestCase{
+				PreCheck:  func() { skipUnlessMode(t, mode) },
+				Providers: testAccProviders,
+				Steps: []resource.TestStep{
+					{
+						Config: config,
+						Check:  resource.ComposeTestCheckFunc(),
+					},
+					{
+						Config: config2,
+						Check:  check,
+					},
+				},
+			})
+		}
+
+		t.Run("with an anonymous account", func(t *testing.T) {
+			t.Skip("anonymous account not supported for this operation")
+		})
+
+		t.Run("with an individual account", func(t *testing.T) {
+			testCase(t, individual)
```

Comment:
This test is failing for me for an individual account but passing for an organization account. Is it succeeding for you when running locally?

Reply:
Locally it only run with an organization account. And is ok
```
$ TF_ACC=1 go test -v   ./... -run ^TestAccGithubRepositoryBranchesDataSource
?   	github.com/integrations/terraform-provider-github/v5	[no test files]
=== RUN   TestAccGithubRepositoryBranchesDataSource
=== RUN   TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository
=== RUN   TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository/with_an_anonymous_account
    data_source_github_repository_branches_test.go:46: anonymous account not supported for this operation
=== RUN   TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository/with_an_individual_account
    provider_utils.go:56: GITHUB_TOKEN and GITHUB_OWNER environment variables should be set
    provider_utils.go:66: Skipping TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository/with_an_individual_account which requires individual mode
=== RUN   TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository/with_an_organization_account
=== RUN   TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository_with_filtering
=== RUN   TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository_with_filtering/with_an_anonymous_account
    data_source_github_repository_branches_test.go:125: anonymous account not supported for this operation
=== RUN   TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository_with_filtering/with_an_individual_account
    provider_utils.go:56: GITHUB_TOKEN and GITHUB_OWNER environment variables should be set
    provider_utils.go:66: Skipping TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository_with_filtering/with_an_individual_account which requires individual mode
=== RUN   TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository_with_filtering/with_an_organization_account
--- PASS: TestAccGithubRepositoryBranchesDataSource (28.78s)
    --- PASS: TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository (7.96s)
        --- SKIP: TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository/with_an_anonymous_account (0.00s)
        --- SKIP: TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository/with_an_individual_account (0.00s)
        --- PASS: TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository/with_an_organization_account (7.96s)
    --- PASS: TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository_with_filtering (20.82s)
        --- SKIP: TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository_with_filtering/with_an_anonymous_account (0.00s)
        --- SKIP: TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository_with_filtering/with_an_individual_account (0.00s)
        --- PASS: TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository_with_filtering/with_an_organization_account (20.82s)
PASS
ok  	github.com/integrations/terraform-provider-github/v5/github	29.454s
```

Reply:
Individal test is ok too:

```
$ TF_ACC=1 go test -v   ./... -run ^TestAccGithubRepositoryBranchesDataSource
?   	github.com/integrations/terraform-provider-github/v5	[no test files]
=== RUN   TestAccGithubRepositoryBranchesDataSource
=== RUN   TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository
=== RUN   TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository/with_an_anonymous_account
    data_source_github_repository_branches_test.go:46: anonymous account not supported for this operation
=== RUN   TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository/with_an_individual_account
=== RUN   TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository/with_an_organization_account
=== RUN   TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository_with_filtering
=== RUN   TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository_with_filtering/with_an_anonymous_account
    data_source_github_repository_branches_test.go:125: anonymous account not supported for this operation
=== RUN   TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository_with_filtering/with_an_individual_account
=== RUN   TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository_with_filtering/with_an_organization_account
--- PASS: TestAccGithubRepositoryBranchesDataSource (55.48s)
    --- PASS: TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository (15.47s)
        --- SKIP: TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository/with_an_anonymous_account (0.00s)
        --- PASS: TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository/with_an_individual_account (7.76s)
        --- PASS: TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository/with_an_organization_account (7.71s)
    --- PASS: TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository_with_filtering (40.01s)
        --- SKIP: TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository_with_filtering/with_an_anonymous_account (0.00s)
        --- PASS: TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository_with_filtering/with_an_individual_account (19.50s)
        --- PASS: TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository_with_filtering/with_an_organization_account (20.51s)
PASS
ok  	github.com/integrations/terraform-provider-github/v5/github	56.002s

Reply:
Weird...I'm getting errors when I run both (organization and individual) at the same time, but both tests are passing individually.